File: data/EXP01/python/2502/2502.output_3.py
# @lc app=leetcode id=2502 lang=python3
class Allocator:

    def __init__(self, n: int):
        self.memory = [None] * n

    def allocate(self, size: int, mID: int) -> int:
        start = -1
        count = 0
        for i in range(len(self.memory)):
            if self.memory[i] is None:
                if count == 0:
                    start = i
                count += 1
                if count == size:
                    for j in range(start, start + size):
                        self.memory[j] = mID
                    return start
            else:
                count = 0
        return -1

    def free(self, mID: int) -> int:
        count = 0
        for i in range(len(self.memory)):
            if self.memory[i] == mID:
                self.memory[i] = None
                count += 1
        return count

File: data/EXP01/go/2502/2502.output_3.go
type Allocator struct {
    memory []int
}

func Constructor(n int) Allocator {
    return Allocator{
        memory: make([]int, n),
    }
}

func (this *Allocator) Allocate(size int, mID int) int {
    for i := 0; i <= len(this.memory)-size; i++ {
        available := true
        for j := i; j < i+size; j++ {
            if this.memory[j] != 0 {
                available = false
                break
            }
        }
        if available {
            for j := i; j < i+size; j++ {
                this.memory[j] = mID
            }
            return i
        }
    }
    return -1
}

func (this *Allocator) Free(mID int) int {
    count := 0
    for i := 0; i < len(this.memory); i++ {
        if this.memory[i] == mID {
            this.memory[i] = 0
            count++
        }
    }
    return count
}


